### PR TITLE
[TGUI] Mob Spawner Menu

### PR DESCRIPTION
**Pull request by TDSSS**
## What Does This PR Do
This PR converts the existing mob spawner menu to use TGUI. It also means that a mob spawner can now have 3 kinds of info:
Description: An OOC description of what this role is about, telling you what exactly  the role is expected to do and be about.
Fluff: An IC description to help you get into your role.
Important Info: Big, bold and red, this is things like telling ashwalkers not to go to the station without permission.

Fluff and Important Info are optional, description is not.

Some of the descriptions/fluff could probably use some work. For example, I am not sure what the vet is even supposed to do, so the OOC explanation text for it is a bit weird.

## Why It's Good For The Game
First, separating IC description and OOC description is good in general for telling players what to expect from a role. it also allows us to highlight important info/rules associated with a role easier.

Secondly, TGUI good.

## Images of changes
Updated look:
Untitled

Here's the UI as it currently looks. Keep in mind in a normal round there will be less total spawners.

## Changelog
:cl:
tweak: ghost role spawner menu now uses TGUI
/:cl:

### CODE REVIEW

**Comment by TDSSS** [comment]
I've updated this with a few changes. Linebreaks are now respected and fluff is marked as not too important with a grey colour. With that I think I think this is no longer a draft.

**Comment by dearmochi** [comment]
TGUI part looks good!

File: code/game/gamemodes/miniantags/bot_swarm/swarmer.dm [hunk 1]
```diff
@@ -19,15 +19,14 @@
...
-	<b>Clicking on any object will try to consume it, either deconstructing it into its components, destroying it, or integrating any materials it has into you if successful.</b>
-	<b>Ctrl-Clicking on a mob will attempt to remove it from the area and place it in a safe environment for storage.</b>
-	<b>Objectives:</b>
+	important_info = "Follow your objectives, do not make the station inhospitable or try and kill crew."
+	//TODO line breaks aren't carried through to the description, meaning the objectives are all just in text rather than on their own lines. Need to figur eout how to fix that.
+	flavour_text = "You are a swarmer, a weapon of a long dead civilization. Until further orders from your original masters are received, you must continue to consume and replicate."
+	description = {" Your goal is to create more of yourself by consuming the station. Clicking on any object will try to consume it, either deconstructing it into its components, destroying it, or integrating any materials it has into you if successful. Ctrl-Clicking on a mob will attempt to remove it from the area and place it in a safe environment for storage.
```
Comment (AffectedArc07) [inline code/game/gamemodes/miniantags/bot_swarm/swarmer.dm:25]: If line breaks arent supported, why are some still in here

File: code/game/gamemodes/miniantags/bot_swarm/swarmer.dm [hunk 2]
```diff
@@ -19,15 +19,14 @@
 	mob_name = "a swarmer"
 	death = FALSE
 	roundstart = FALSE
-	flavour_text = {"
-	<b>You are a swarmer, a weapon of a long dead civilization. Until further orders from your original masters are received, you must continue to consume and replicate.</b>
-	<b>Clicking on any object will try to consume it, either deconstructing it into its components, destroying it, or integrating any materials it has into you if successful.</b>
-	<b>Ctrl-Clicking on a mob will attempt to remove it from the area and place it in a safe environment for storage.</b>
-	<b>Objectives:</b>
+	important_info = "Follow your objectives, do not make the station inhospitable or try and kill crew."
```
Comment (AffectedArc07) [inline code/game/gamemodes/miniantags/bot_swarm/swarmer.dm:22]: Can you add a comment to show where each of these will appear, like mention if important_info is the bold red one thats for OOC rules, etc

File: code/game/gamemodes/miniantags/bot_swarm/swarmer.dm (hunk 1 above)
Comment (TDSSS) [reply code/game/gamemodes/miniantags/bot_swarm/swarmer.dm:25]: Comment is old, I figured out how to get linebreaks to work. Will fix shortly

File: code/game/gamemodes/miniantags/bot_swarm/swarmer.dm [hunk 3]
```diff
@@ -111,7 +110,7 @@
 	to_chat(src, "<b>You are a swarmer, a weapon of a long dead civilization. Until further orders from your original masters are received, you must continue to consume and replicate.</b>")
 	to_chat(src, "<b>Clicking on any object will try to consume it, either deconstructing it into its components, destroying it, or integrating any materials it has into you if successful.</b>")
 	to_chat(src, "<b>Ctrl-Clicking on a mob will attempt to remove it from the area and place it in a safe environment for storage.</b>")
-	to_chat(src, "<b>Prime Directives:</b>")
+	to_chat(src, "<b>Prime Directives:")
```
Comment (AffectedArc07) [inline code/game/gamemodes/miniantags/bot_swarm/swarmer.dm:113]: Any reason you removed the bold closing tag here

Comment (TDSSS) [reply code/game/gamemodes/miniantags/bot_swarm/swarmer.dm:113]: Collateral damage from removing the tags from the description of the spawner. My bad.

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**